views/js/pciCreator/dev/audioRecordingInteraction/runtime/js/recorder.js

```diff
@@ -107,6 +118,42 @@ define([
             analyser,               // the WebAudio node use to read the input level
             frequencyArray;         // used to compute the input level from the current array of frequencies
 
+        /**
+         * Check if client uses the iOS device.
+         *
+         * @returns {*|boolean}
+         */
+        function isIOSDevice() {
+            return /(iPhone|iPad)/i.test(navigator.userAgent)
+        }
+
+        /**
+         * Fill the frequencyArray with fake data
+         *
+         * @param {Array} frequencyArray
+         */
+        function fillFakeEmitter(frequencyArray) {
+            var frequencyLength = frequencyArray.length;
+            var level = 0;
+            var levelIndex = 0;
+
+            if (frequencyLength) {
+                level = getRndInteger(frequencyLength * (frequencyLength / 2), frequencyLength * frequencyLength);
+            }
+
+            while (levelIndex++ < frequencyLength) {
+                frequencyArray[levelIndex] = level;
+                if (frequencyLength / 2 < levelIndex) {
+                    /**
+                     * As far as we're imitating a fake icon with bouncing sound levels, so it would be better
+                     * to fill only the half of `frequencyArray`, which represents the scale for levels of our icon.
+                     * If we fill the whole array, then it will seem that sound is too loud and levels almost maximal.
+                     */
+                    break;
```

Comment:
Having a `break` in a loop should be avoided whenever possible. This introduces an unnecessary complexity, and could lead to issues.
Instead, for better readability, and if you need the action being done being the condition is evaluated, you should have done something like:
```
do {
    frequencyArray[levelIndex++] = level;
} while (levelIndex < frequencyLength / 2);
```

Please also note, with your construction, the first index will never be filled, as the index will be incremented before entering the loop.

Reply:
Applied you advices except the `postfix increment` operator.